smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt

```diff
@@ -114,11 +111,23 @@ class MiddlewareExecutionGenerator(
             writer.write("  .withExpiration(value: expiration)")
         }
 
+        // Add context values for config fields
+        val configProperties = ctx.integrations
+            .flatMap { it.clientConfigurations(ctx) }
+            .flatMap { it.getProperties(ctx) }
+        configProperties.forEach {
+            renderConfigPropertyToContext(it)
+        }
```

Comment:
The block of code ahead iterates over config fields, to see if any should be written to the context.

This is useful for reporting various config options in the user-agent later.